lib/openhab/dsl/rules/triggers/trigger.rb

```diff
@@ -61,9 +61,9 @@ def append_trigger(type, config)
         # @return [Array] A new flat array with any GroupMembers object left intact
         #
         def separate_groups(item_array)
-          return item_array if item_array.grep(Array).length.zero?
+          groups, items = item_array.partition { |item| item.is_a?(OpenHAB::DSL::Items::GroupItem::GroupMembers) }
```

Comment:
and now I see with the other failing test you're trying to accept a regular Array here too. will fix to accept anything that's Enumerable (which will then still allow GroupMembers to get properly separated, even though it's now an Enumerable, but does not act like an Array)